backend/src/util/VerifyFIN.ts

```diff
@@ -0,0 +1,34 @@
+const checkAlpha = ['X', 'W', 'U', 'T', 'R', 'Q', 'P', 'N', 'M', 'L', 'K'];
+const weightArray = [2, 7, 6, 5, 4, 3, 2];
+
+function validateFIN(fin: string): string {
+  const finUpper = fin.toUpperCase();
+  if (finUpper.length === 9) {
+    const sumArray = fin.slice(1, 8).split('').map(Number); // FIN String into array of numbers to multiply by weight
+
+    let sum = 0;
+    for (let i = 0; i < sumArray.length; i += 1) {
+      sum += sumArray[i] * weightArray[i];
+    } // Cross multiply with weightArray to get sum
+
+    if (finUpper[0] === 'G') {
+      sum += 4;
+    } // FIN that starts with G has their checkAlpha shifted by 4
+
+    if (checkAlpha[sum % 11] === finUpper.slice(-1)) {
+      return `${fin} is a valid FIN.`;
+    }
+    return `${fin} is an invalid FIN.`;
+  }
+  return 'Wrong Format.';
+}
+
+export default validateFIN;
+
+// Tests
```

Comment:
Hey. Are you familiar with jest? would be good to have these guys as unit tests since you already have the scenarios.

Reply:
Not really but I will look it up!

Reply:
Agree. It might be a bit difficult to set up Jest from scratch, but I think it's a good exercise for you @fabianchua6. No pressure, I can come in and help you perfect it. Just try to get the following:

- install jest
- write at least one test case
- run the tests